public/app/features/explore/Logs/Logs.tsx

```diff
@@ -156,81 +141,90 @@ const getDefaultVisualisationType = (): LogsVisualisationType => {
-  }
-
-  componentWillUnmount() {
-    if (this.flipOrderTimer) {
-      window.clearTimeout(this.flipOrderTimer);
-    }
-
-    if (this.cancelFlippingTimer) {
-      window.clearTimeout(this.cancelFlippingTimer);
-    }
-
-    // If we're unmounting logs (e.g. switching to another datasource), we need to remove the table specific panel state, otherwise it will persist in the explore url
-    if (
-      this.props?.panelState?.logs?.columns ||
-      this.props?.panelState?.logs?.refId ||
-      this.props?.panelState?.logs?.labelFieldName
-    ) {
-      dispatch(
-        changePanelState(this.props.exploreId, 'logs', {
-          ...this.props.panelState?.logs,
-          columns: undefined,
-          visualisationType: this.state.visualisationType,
-          labelFieldName: undefined,
-          refId: undefined,
-        })
-      );
-    }
-  }
-
-  registerLogLevelsWithContentOutline = () => {
+const UnthemedLogs: React.FunctionComponent<Props> = (props: Props) => {
+  const {
+    width,
+    splitOpen,
+    logRows,
+    logsMeta,
+    logsVolumeEnabled,
+    logsVolumeData,
+    loadLogsVolumeData,
+    loading = false,
+    onClickFilterLabel,
+    onClickFilterOutLabel,
+    timeZone,
+    scanning,
+    scanRange,
+    showContextToggle,
+    absoluteRange,
+    onChangeTime,
+    getFieldLinks,
+    theme,
+    logsQueries,
+    clearCache,
+    addResultsToCache,
+    exploreId,
+    getRowContext,
+    getLogRowContextUi,
+    getRowContextQuery,
+    loadMoreLogs,
+    panelState,
+    eventBus,
+  } = props;
+  const [showLabels, setShowLabels] = useState<boolean>(store.getBool(SETTINGS_KEYS.showLabels, false));
+  const [showTime, setShowTime] = useState<boolean>(store.getBool(SETTINGS_KEYS.showTime, true));
+  const [wrapLogMessage, setWrapLogMessage] = useState<boolean>(store.getBool(SETTINGS_KEYS.wrapLogMessage, true));
+  const [prettifyLogMessage, setPrettifyLogMessage] = useState<boolean>(
+    store.getBool(SETTINGS_KEYS.prettifyLogMessage, false)
+  );
+  const [dedupStrategy, setDedupStrategy] = useState<LogsDedupStrategy>(LogsDedupStrategy.none);
+  const [hiddenLogLevels, setHiddenLogLevels] = useState<LogLevel[]>([]);
+  const [logsSortOrder, setLogsSortOrder] = useState<LogsSortOrder>(
+    store.get(SETTINGS_KEYS.logsSortOrder) || LogsSortOrder.Descending
+  );
+  const [isFlipping, setIsFlipping] = useState<boolean>(false);
+  const [displayedFields, setDisplayedFields] = useState<string[]>([]);
+  const [forceEscape, setForceEscape] = useState<boolean>(false);
+  const [contextOpen, setContextOpen] = useState<boolean>(false);
+  const [contextRow, setContextRow] = useState<LogRowModel | undefined>(undefined);
+  const [visualisationType, setVisualisationType] = useState<LogsVisualisationType | undefined>(
+    panelState?.logs?.visualisationType ?? getDefaultVisualisationType()
+  );
+  const [logsContainer, setLogsContainer] = useState<HTMLDivElement | undefined>(undefined);
```

Comment:
This doesn't need to be a state variable, and also by the way it's implemented we can just use a ref.

```
const onLogsContainerRef = useCallback((node: HTMLDivElement) => {
    setLogsContainer(node);
  }, []);
```

Reply:
Addressed in https://github.com/grafana/grafana/pull/87808/commits/55b82654ab0c6dbffef5a3164c486829a288a301